Mage/src/main/java/mage/abilities/effects/ContinuousEffectImpl.java

```diff
@@ -515,4 +516,12 @@ public boolean canLookAtNextTopLibraryCard(Game game) {
         }
         return true;
     }
+
+    public boolean hasHint() {
+        return false;
+    }
+
+    public String getHint(Permanent permanent, Ability source, Game game) {
```

Comment:
Return real hint objects (multiple) here, not text only.

Reply:
Yes, that makes sense. I didn't do it originally because I wasn't using `Ability` but it turns out you need it anyway. Let me work on it.

Reply:
Also must rename that method like getAffectedHints and make sure it all that hints compatible with hints tool (see player panel). If it (tool) generates wrongly or dirty data then report here with examples — can be improved.